pkg/command/eventing/driver/kafka_sender.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/Shopify/sarama"
+	cloudeventskafka "github.com/cloudevents/sdk-go/protocol/kafka_sarama/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type KafkaEventSender struct {
+	Plan SendEventsPlan
+	//SleepTimeSeconds float64 // simulate sending time
+}
+
+func (s KafkaEventSender) Send() EventsStats {
+	plan := s.Plan
+	//eventsSent := int(float64(plan.eventsPerSecond) * plan.durationSeconds)
+	stats := sendEvents(plan)
+	// TODO better golang idiom/conversion?
+	// sleepTimeNano := int64(plan.durationSeconds * float64(time.Second))
+	// time.Sleep(time.Duration(sleepTimeNano))
+	// timeSeconds := plan.durationSeconds
+	//stats := EventsStats{plan.senderName, eventsSent, timeSeconds, 0, 0}
+	return stats
+}
+
+type envConfig struct {
+	// Port on which to listen for cloudevents
+	//Port int `envconfig:"PORT" default:"8080"`
+
+	// KafkaServer URL to connect to the Kafka server.
+	KafkaServer string `envconfig:"KAFKA_BOOTSTRAP_SERVERS" required:"true"`
+
+	// Subject is the nats subject to publish cloudevents on.
+	Topic string `envconfig:"KAFKA_TOPIC" required:"true"`
+}
+
+func sendEvents(plan SendEventsPlan) EventsStats {
```

Comment:
Break this down into smaller helper function and that will allow you to improve and better visualize this code compared to the Redis sender and others. Also, makes it easier to debug and review :)